### ISSUE TITLE
About curvature filtering

### ISSUE THREAD

**Issue by lilipopololo** [TruongKhang/cds-mvsnet#5]
Hello, author, I want to ask about the curvature filtering
（1）The role of curvature filtering in your code ，What  is the Norm Curv (NC) in the code? What is the mean of Norm Curv.
（2）Why do you need to compute Epipolar after computing fundamental matrix in your code（and Epipolar is Epipolar line or Epipolar point? Whose Epipolar point?）
(3)In the following code, the x and y coordinates in the feature map are converted to the coordinates centered on epipolar point. Can I understand it this way？

```
#         u = x.unsqueeze(0).unsqueeze(0) - epipole_map[:, [0], :, :] # [B, 1, H, W]
#         v = y.unsqueeze(0).unsqueeze(0) - epipole_map[:, [1], :, :] # [B, 1, H, W]
```

Thank you for your reply and your perfect work
(and i train you code on dtu for 3o epock,but got 0.36 and 0.28

**Comment by lilipopololo** [comment]
0.36 and 0.28 due to I train this model on one card I believe

**Comment by TruongKhang** [comment]
hiiiiii @lilipopololo, 
(1) the main role of `norm_curv` is to regularize the curvature values. It should be restricted to a small range. If you take a look at our paper. We explained this in Eq. (5) of Section 3.2 and the loss function in Section 4.2.
(2) After computing the fundamental matrix, we need to compute the epipoles (i.e., epipolar points) in both reference and source images. Then, we can estimate the epipolar line for each coordinate (x, y) in the feature map.
(3) The direction of epipolar line for each pixel (x, y) is defined as

    u = x.unsqueeze(0).unsqueeze(0) - epipole_map[:, [0], :, :] # [B, 1, H, W]
    v = y.unsqueeze(0).unsqueeze(0) - epipole_map[:, [1], :, :] # [B, 1, H, W]
Then we normalize (u, v) to obtain an unit vector. This unit vector is the variable $\omega$ in Eq. (5) of our paper.

About the result, did you use settings similar to ours when testing? Also, I think 0.36 and 0.28 are only a small difference.  You can refer to this [config](https://github.com/TruongKhang/cds-mvsnet/blob/main/pretrained/dtu_only/config.json) file. Let me know more detail if you cannot reproduce the results.
